### PR TITLE
v5.17.23

### PR DESCRIPTION
**Pull request by cherniavskii**
- [x] Compare the last tag with the branch upon which you want to release (`next` for the alpha / beta releases and `master` for the current stable version).
      To do so, use `yarn release:changelog` The options are the following:

  ```bash
  yarn release:changelog
     --githubToken   YOUR_GITHUB_TOKEN (needs "public_repo" permission)
     --lastRelease   The release to compare against (default: the last one)
     --release       The branch to release (default: master)
  ```

  You can also provide the github token by setting `process.env.GITHUB_TOKEN` variable.

  In case of a problem, another method to generate the changelog is available at the end of this page.

- [x] Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
- [x] Update the root `package.json`'s version
- [x] Update the versions of the other `package.json` files and of the dependencies with `yarn release:version`.
- [x] Fix manually the package version in `x-date-pickers/package.json` and `x-date-pickers-pro/package.json`.
- [x] Open PR with changes and wait for review and green CI.
- [ ] Merge PR once CI is green, and it has been approved.

### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```sh
git push upstream master:docs-v5 -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v5)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v5` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### Announce

- [ ] Follow the instructions in https://mui-org.notion.site/Releases-7490ef9581b4447ebdbf86b13164272d.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,38 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## v5.17.23
+
+_Feb 9, 2023_
+
+We'd like to offer a big thanks to the 6 contributors who made this release possible. Here are some highlights ✨:
+
+TODO INSERT HIGHLIGHTS
```
Comment (MBilalShafi) [inline CHANGELOG.md:12]:
```suggestion
- 🌍 Improve Brazilian Portuguese (pt-BR) locale
- 🐞 Bugfixes
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,38 @@
...
+
+### `@mui/x-data-grid@v5.17.23` / `@mui/x-data-grid-pro@v5.17.23` / `@mui/x-data-grid-premium@v5.17.23`
+
+#### Changes
+
+- [DataGrid] Allow to customize the value displayed in the filter button (#7816) @ithrforu
```
Comment (MBilalShafi) [inline CHANGELOG.md:18]:
```suggestion
- [DataGrid] Allow to customize the value displayed in the filter button tooltip (#7816) @ithrforu
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,38 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## v5.17.23
```
Comment (LukasTy) [inline CHANGELOG.md:6]: Nitpick 🙈 
```suggestion
## 5.17.23
```
Screenshot 2023-02-09 at 13 41 19
Screenshot 2023-02-09 at 13 41 27

File: CHANGELOG.md (hunk 1 above)
Comment (LukasTy) [reply CHANGELOG.md:12]: Maybe one more place 
```suggestion
- 🌍 Improve Brazilian Portuguese (pt-BR) locale
- 🎉 Add banner and callouts to inform about MUI X v6 beta
- 🐞 Bugfixes
```
Maybe yet another opportunity/place to increase visibility? 🙈

File: packages/grid/x-data-grid-generator/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mui/x-data-grid-generator",
-  "version": "5.17.20",
+  "version": "5.17.21",
```
Comment (LukasTy) [inline packages/grid/x-data-grid-generator/package.json:3]: Shouldn't it's version always be synced with the main release version? 🙈

**Review by LukasTy — APPROVED** [review]

**Pull request merged**